target_tools/thread_manager.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+"""ThreadManager for managing async requests"""
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=no-member
+import atexit
+from multiprocessing.pool import ThreadPool
```

Comment:
It's probably better to use multithreading instead of multiprocessing here, as we're mostly I/O bound when fetching the Geo data, also there's less of an overhead with multithreading